synapse/datamodel.py

```diff
@@ -253,7 +256,10 @@ def __init__(self, modl, name, info):
         self.props = {}     # name: Prop()
         self.refsout = None
 
-        if self.info.get('deprecated') or self.type.deprecated:
+        self.locked = False
+        self.deprecated = self.type.deprecated
```

Comment:
Is this matching form and type deprecation **together**? previously they were separate